server/util/consistent_hash/consistent_hash.go

```diff
@@ -58,3 +63,48 @@ func (c *ConsistentHash) Get(key string) string {
 	}
 	return c.ring[c.keys[idx]]
 }
+
+func (c *ConsistentHash) GetNReplicas(key string, n int) []string {
+	c.mu.RLock()
+	defer c.mu.RUnlock()
+	if len(c.keys) == 0 {
+		return nil
+	}
+	h := c.hashKey(key)
+	idx := sort.Search(len(c.keys), func(i int) bool {
+		return c.keys[i] >= h
+	})
+	if idx == len(c.keys) {
+		idx = 0
+	}
+	replicas := make([]string, 0, n)
+	seen := make(map[string]struct{}, 0)
+
+	original := c.ring[c.keys[idx]]
+	replicas = append(replicas, original)
+	seen[original] = struct{}{}
+
+	for offset := 1; offset < len(c.keys); offset += 1 {
+		newIdx := (idx + offset) % len(c.keys)
+		v := c.ring[c.keys[newIdx]]
+		if _, ok := seen[v]; !ok {
+			replicas = append(replicas, v)
+			seen[v] = struct{}{}
+		}
+		if len(replicas) == n {
+			break
+		}
+	}
+	if len(replicas) < n {
+		log.Printf("Warning: client requested %d replicas but only %d were available.", n, len(replicas))
+	}
+	return replicas
+}
+
+func (c *ConsistentHash) GetNext(key string) string {
```

Comment:
docs pls?

Reply:
(Removed, but added docs on the two above.)